sdk/core/Azure.Core/tests/TestFramework/TestEnvironment.cs

```diff
@@ -0,0 +1,50 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using Azure.Identity;
+
+namespace Azure.Core.Testing
+{
+    public class TestEnvironment
```

Comment:
If these are to be public, might be good to put comments on members, especially what to expect with some of the methods' behavior below.